packages/gp2-components/src/organisms/ResourceModal.tsx

```diff
@@ -38,23 +39,54 @@ const overrideButtonStyles = css({
 });
 
 type ResourceModalProps = Partial<gp2.Resource> & {
+  modalTitle: string;
+  modalDescription: string;
   backHref: string;
   onSave?: (data: gp2.Resource) => void | Promise<void>;
 };
 
-const ResourceModal: React.FC<ResourceModalProps> = (props) => {
-  const { type, title, description, backHref, onSave = noop } = props;
-  const externalLink = type === 'Link' && props.externalLink;
-  const [newType, setNewType] = useState<'Link' | 'Note' | ''>(type || '');
+const ResourceModal: React.FC<ResourceModalProps> = ({
+  title,
+  description,
+  backHref,
+  modalTitle,
+  modalDescription,
+  onSave = noop,
+  ...props
+}) => {
+  const externalLink = props.type === 'Link' && props.externalLink;
+  const [newType, setNewType] = useState<'Link' | 'Note' | ''>(
+    props.type || '',
+  );
   const [newTitle, setNewTitle] = useState(title || '');
   const [newDescription, setNewDescription] = useState(description || '');
   const [newExternalLink, setNewExternalLink] = useState(externalLink || '');
 
+  const verifyDirty = () => {
+    if (props.type === undefined && newType === '') {
+      return false;
+    }
+
+    if (!externalLink && newExternalLink.length === 0) {
```

Comment:
is this condition not covered on the one bellow?

Reply:
test on the type